src/cryptonote_core/service_node_list.cpp

```diff
@@ -312,7 +320,114 @@ namespace service_nodes
 
 
     m_rollback_events.push_back(std::unique_ptr<rollback_event>(new rollback_change(block_height, key, iter->second)));
+    uint64_t swarm_id = iter->second.swarm_id;
     m_service_nodes_infos.erase(iter);
+    check_and_process_swarm_decomission(swarm_id);
+  }
+
+  void service_node_list::check_and_process_swarm_decomission(uint64_t swarm_id)
+  {
+    size_t count = 0;
+    for (const auto& entry : m_service_nodes_infos)
+      if (entry.second.swarm_id == swarm_id)
+        count++;
```

Comment:
Worth turning into a function

Reply:
you know it takes time to read these comments right?

Reply:
Well .. this is a code review :/

Reply:
Fair enough

Reply:
changed to get_swarm_size(...)

Reply:
> code review

The 

> Well .. this is a code review :/

The point here is that addressing small parts of code and being too nitty can end up costing lots of time.

This issue in question, I opted to do what you want, instead of explaining the reason for my objection, since it costs so much time.

The reason for the objection is this code doesn't repeat anywhere, from memory (I haven't checked, and it may well do that now).

The code clarity to be gained from turning this into a function isn't so big, hence why the comment this is perhaps too nitty.

I invited you to lunch / dinner sometime to catch up. I'm interested how your experience was last year, but I don't know if you got the email - are you still doyle@loki.network?

Reply:
Hi, long time no see. I've found your email, it was buried in my inbox.

Reply:
> buried in my inbox

You mean, you _ignored it_. =b

It's okay. Let's catch up sometime soon though.